client/components/AdoptionForm.js

```diff
@@ -0,0 +1,98 @@
+import React,{useState} from "react"
+
+const AdoptionForm =()=>{
+const[adoptionData, setAdoptionData] = useState({
+  name : "",
+  phoneNumber : "",
+  email: "",
+  homeStatus: "own"
+})
+let errorContainer = null
+const [errors, setErrors]=useState([])
+if(errors.length!==0){
+  errorContainer= errors.map(error=>{
+    return <p>{error}</p>
+  })
+}
+
+const handleChange =(event)=>{
+setAdoptionData({
+  ...adoptionData,
+  [event.currentTarget.name]: event.currentTarget.value
+})
+}
+
+const checkForErrors =()=>{
+  const errorsArray = []
+  const fieldsData = Object.keys(adoptionData)
+  fieldsData.forEach(fieldName=>{
+    if(adoptionData[fieldName]=== ""){
+    errorsArray.push(fieldName)
+    }
+  })
+  return errorsArray
+}
+
+const handleSubmit = async (event)=>{
+  event.preventDefault()
+  let errorsArray=checkForErrors()
+ if(errorsArray.length===0){
+   await addInfo()
+  setAdoptionData({
+    name : "",
+    phoneNumber : "",
+    email: "",
+    homeStatus: "own"
+   })
+ }
+ else{
+  setErrors(errorsArray)
+ }
+}
+
+const addInfo = async ()=>{
+  try{
+    const response = await fetch("/api/v1/petTypes", {
+      method: "POST",
+      headers: new Headers({
+        "Content-Type" : "application/json"
+      }),
+      body: JSON.stringify(adoptionData)
+    })
+
```

Comment:
Looks like we might be missing some error handling/logic after the response comes through here